src/PrestaShopBundle/Command/SecurityAnnotationLinterCommand.php

```diff
@@ -27,35 +27,155 @@
+ *
+ * @see \PrestaShopBundle\Security\Annotation\AdminSecurity
  */
 final class SecurityAnnotationLinterCommand extends ContainerAwareCommand
 {
+    const ACTION_LIST_ALL = 'list';
+    const ACTION_FIND_MISSING = 'find-missing';
+
+    /**
+     * @param string $expression
+     *
+     * @return string
+     */
+    public static function parseExpression($expression)
+    {
+        $pattern1 = '#\[(.*)\]#';
+        $pattern2 = '#is_granted\((.*),#';
+        $matches1 = [];
+        $matches2 = [];
+        preg_match($pattern1, $expression, $matches1);
+        preg_match($pattern2, $expression, $matches2);
+
+        if (count($matches1) > 1) {
+            return $matches1[1];
+        }
+        if (count($matches2) > 1) {
+            return $matches2[1];
+        }
+
+        return '';
+    }
+
+    /**
+     * @return string[]
+     */
+    public static function getAvailableActions()
+    {
+        return [self::ACTION_LIST_ALL, self::ACTION_FIND_MISSING];
+    }
+
     /**
      * {@inheritdoc}
      */
     public function configure()
     {
+        $description = 'Checks if Back Office route controllers has configured Security annotations.';
+        $actionDescription = sprintf(
+            'Action to perform, must be one of: %s',
+            implode(', ', self::getAvailableActions())
+        );
+
         $this
             ->setName('prestashop:linter:security-annotation')
-            ->setDescription('Checks if Back Office route controllers has configured Security annotations.')
-        ;
+            ->setDescription($description)
+            ->addArgument('action', InputArgument::REQUIRED, $actionDescription);
     }
 
     /**
      * {@inheritdoc}
      */
     protected function execute(InputInterface $input, OutputInterface $output)
     {
-        $adminRouteProvider = $this->getContainer()->get('prestashop.bundle.routing.linter.admin_route_provider');
-        $securityAnnotationLinter = $this->getContainer()
+        $actionToPerform = $input->getArgument('action');
+
+        if (!in_array($actionToPerform, self::getAvailableActions())) {
+            throw new \InvalidArgumentException(sprintf(
+                    'Action must be one of: %s',
+                    implode(', ', self::getAvailableActions())
+                )
+            );
+        }
+
+        switch ($actionToPerform) {
+            case self::ACTION_LIST_ALL:
+                $this->listAllRoutesAndTheirPermissions($input, $output);
+                break;
+            case self::ACTION_FIND_MISSING:
+                $this->findRoutesWithMissingSecurityAnnotations($input, $output);
+                break;
+
+            default:
+                throw new \RuntimeException(sprintf('Unknown action %s', $actionToPerform));
+        }
+    }
+
+    /**
+     * @param InputInterface $input
+     * @param OutputInterface $output
+     */
+    private function listAllRoutesAndTheirPermissions(InputInterface $input, OutputInterface $output)
```

Comment:
```suggestion
    private function listAllRoutesAndRelatedPermissions(InputInterface $input, OutputInterface $output)
```

Reply:
Thanks, applied in https://github.com/PrestaShop/PrestaShop/pull/18235/commits/001e445569e9009913564470183c238688b26456